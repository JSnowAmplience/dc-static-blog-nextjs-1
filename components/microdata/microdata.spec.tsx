import renderer from 'react-test-renderer';
import Microdata from './microdata';

describe('Microdata', (): void => {
  it('should render a full Microdata component', (): void => {
    const props = {
      headline: 'test-headline',
      imageUrl: '//test-imageurl/image-test.jpg',
      authors: [{ name: 'test-author-1' }],
      datePublished: '2019-07-07'
    };
    const wrapper = renderer.create(<Microdata {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Microdata component when author is not defined', (): void => {
    const props = {
      headline: 'test-headline',
      imageUrl: '//test-imageurl/image-test.jpg',
      authors: [{ name: 'test-author-1' }],
      datePublished: '2019-07-07'
    };
    delete props.authors;
    const wrapper = renderer.create(<Microdata {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
