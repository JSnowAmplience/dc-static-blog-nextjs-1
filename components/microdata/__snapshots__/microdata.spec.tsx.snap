// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Microdata should render Microdata component when author is not defined 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"BlogPosting\\",\\"headline\\":\\"test-headline\\",\\"image\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":[\\"//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=16:9\\",\\"//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=4:3\\",\\"//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=1:1\\"]},\\"author\\":{\\"@type\\":\\"Person\\",\\"name\\":[]},\\"datePublished\\":\\"2019-07-07\\",\\"publisher\\":{\\"@type\\":\\"Organization\\",\\"name\\":\\"Amplience\\",\\"logo\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":\\"//i1.adis.ws/i/ampcomcms/amplience-logo-hidef?$poi$&aspect=1:1\\"}}}",
    }
  }
  type="application/ld+json"
/>
`;

exports[`Microdata should render a full Microdata component 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"BlogPosting\\",\\"headline\\":\\"test-headline\\",\\"image\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":[\\"//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=16:9\\",\\"//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=4:3\\",\\"//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=1:1\\"]},\\"author\\":{\\"@type\\":\\"Person\\",\\"name\\":[\\"test-author-1\\"]},\\"datePublished\\":\\"2019-07-07\\",\\"publisher\\":{\\"@type\\":\\"Organization\\",\\"name\\":\\"Amplience\\",\\"logo\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":\\"//i1.adis.ws/i/ampcomcms/amplience-logo-hidef?$poi$&aspect=1:1\\"}}}",
    }
  }
  type="application/ld+json"
/>
`;
